src/Altinn.App.Core/Internal/Process/IProcessEventDispatcher.cs

```diff
@@ -0,0 +1,23 @@
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.App.Core.Internal.Process;
+
+/// <summary>
+/// Interface for dispatching events that occur during a process
+/// </summary>
+public interface IProcessEventDispatcher
+{
+    /// <summary>
+    /// Updates the instance process in storage and dispatches instance events
+    /// </summary>
+    /// <param name="instance">The instance with updated process</param>
+    /// <param name="prefill">Prefill data</param>
+    /// <param name="events">Events that should be dispatched</param>
+    /// <returns>Instance from storage after update</returns>
+    Task<Instance> UpdateProcessAndDispatchEvents(Instance instance, Dictionary<string, string>? prefill, List<InstanceEvent>? events);
+    /// <summary>
+    /// Dispatch events for instance to the events system if AppSettings.RegisterEventsWithEventsComponent is true
+    /// </summary>
+    /// <param name="instance">The instance to dispatch events for</param>
+    Task RegisterEventWithEventsComponent(Instance instance);
```

Comment:
DispatchEvent?

Reply:
Rename to `DispatchEventsToEventsComponent`? As we still have events being dispatched to Storage we need to specify that this is events to the eventscomponent I think